### DISCUSSION TITLE
Questions About Terminus Installation on Windows

### DISCUSSION THREAD

**Discussion by husniadil** [beclab/Olares#390] [Q&A] [answered]
Hi guys,

I successfully installed Terminus on Windows and have a few questions.

#### Regarding Terminus Name

I registered Terminus using the default domain (myterminus.com) and verified it using my **@gmail.com** account. If I decide to create a Terminus instance using my **actual domain** in the future, what should I do? Should I reinstall Terminus and register with a different email on Termipass?

#### Regarding Public Domain

I was able to work within my local network and tried using the VPN feature on Termipass. This allowed me to work with files/documents while connected to a completely different network. However, I couldn’t access the full desktop UI of Terminus from outside my network. I attempted to make it public via Cloudflare, but when I accessed the domain, it didn’t display the expected UI. Instead, I encountered a blank page with a link to [Bytetrade](https://www.bytetradelab.io/). How can I properly access the desktop UI while on mobile?

#### Regarding Development

Given that Terminus uses Kubernetes under the hood, I plan to install Rancher Desktop. Will this be compatible? Will they run in the same local Kubernetes cluster?

#### Regarding DNS Propagation

Lastly, I'm curious about how my router recognizes **desktop.local.\<terminus-name\>.myterminus.com** and other subdomains such as **market.local.\<terminus-name\>.myterminus.com** without these being specified in the router settings. I checked both the /etc/hosts and the /mnt/c/Windows/System32/drivers/etc/hosts files, but I didn't find any entries for the myterminus domain. Can you help to briefly explain what's going on behind the scene?

Thank you!

**Comment by TShentu** [comment, marked as answer]
Thank you for your interest in Terminus! 

1. Regarding Using Your Own Domain
Yes, you’ll need to [apply for a Custom Domain](https://docs.jointerminus.com/how-to/space/domain/host-domain.html) first. After that, reinstall and activate your Terminus with your Terminus Name on the custom domain. For detailed setup instructions, please refer to the [Organization Name](https://docs.jointerminus.com/developer/contribute/snowinning/terminus-name.html#organization) section.

2. Accessing Terminus from Outside the Network
  If your device and Terminus are on different networks:
    - On Windows, install the Windows version of TermiPass, import your account using your mnemonic, and enable the Tailscale feature.
   - For mobile access, the same steps apply.


3. Regarding Development with Rancher Desktop
Our team hasn’t specifically tested compatibility with Rancher Desktop. However, since we haven't modified Kubernetes, it should work. We recommend trying our built-in app, ControlHub, which is an enhanced version of the Kubesphere Console designed for container management.

4. About DNS Propagation
  For each Terminus Name, we configure two types of DNS resolutions:
    - _`terminus-name`_.myterminus.com: This resolves to a public IP address, which could be the address of a Cloudflare tunnel or an FRP server.
    - local._`terminus-name`_.myterminus.com: This resolves to the machine's internal IP address, such as 192.168.xx.xx.

Hope I have answered your questions. Let us know if you have further questions.
